mm2src/mm2_main/src/rpc/lp_commands/one_inch/types.rs

```diff
@@ -146,26 +159,59 @@ pub struct ClassicSwapResponse {
     pub gas: Option<u128>,
 }
 
-impl ClassicSwapResponse {
-    pub(crate) fn from_api_classic_swap_data(
-        data: one_inch_api::types::ClassicSwapData,
-        decimals: u8,
+/// Response for both classic swap quote or create swap calls
+pub type ClassicSwapResponse = ClassicSwapDetails;
+
+impl ClassicSwapDetails {
+    /// Get token name as it is defined in the coins file by contract address  
+    async fn token_name_kdf(ctx: &MmArc, chain_id: u64, token_info: &LrTokenInfo) -> Option<Ticker> {
+        let special_contract =
+            Address::from_str(ApiClient::eth_special_contract()).expect("1inch special address must be valid"); // TODO: must call 1inch to get it, instead of burned consts
+
+        let platform_ticker = get_platform_ticker(ctx, chain_id)?;
+        if token_info.address == special_contract {
+            Some(platform_ticker)
+        } else {
+            get_erc20_ticker_by_contract_address(ctx, &platform_ticker, &token_info.address)
+        }
+    }
+
+    pub(crate) async fn from_api_classic_swap_data(
+        ctx: &MmArc,
+        chain_id: u64,
+        data: one_inch_api::classic_swap_types::ClassicSwapData,
     ) -> MmResult<Self, FromApiValueError> {
+        let src_token_info = data.src_token.ok_or(FromApiValueError("No token info".to_owned()))?;
+        let dst_token_info = data.dst_token.ok_or(FromApiValueError("No token info".to_owned()))?;
+        let src_decimals: u8 = src_token_info
+            .decimals
+            .try_into()
+            .map_to_mm(|_| FromApiValueError("bad decimals".to_owned()))?;
+        let dst_decimals: u8 = dst_token_info
+            .decimals
+            .try_into()
+            .map_to_mm(|_| FromApiValueError("bad decimals".to_owned()))?;
```

Comment:
Error messages "No token info" and "bad decimals" are too generic.
Messages should be updated to clearly specify the context, smth like this:
- "Missing source token info" / "Missing destination token info"
- "Invalid source token decimals: value out of range for u8" / "Invalid destination token decimals: value out of range for u8"

Reply:
fixed error messages